python/repeated.c

```diff
@@ -298,6 +295,69 @@ static PyObject* PyUpb_RepeatedContainer_Subscript(PyObject* _self,
   }
 }
 
+static int PyUpb_RepeatedContainer_SetSubscript(
+    PyUpb_RepeatedContainer* self, upb_array* arr, const upb_fielddef* f,
+    Py_ssize_t idx, Py_ssize_t count, Py_ssize_t step, PyObject* value) {
+  upb_arena* arena = PyUpb_Arena_Get(self->arena);
+  if (upb_fielddef_issubmsg(f)) {
+    PyErr_SetString(PyExc_TypeError, "does not support assignment");
+    return -1;
+  }
+
+  if (step == 0) {
+    // Set single value.
+    upb_msgval msgval;
+    if (!PyUpb_PyToUpb(value, f, &msgval, arena)) return -1;
+    upb_array_set(arr, idx, msgval);
+    return 0;
+  }
+
+  // Set range.
+  PyObject* seq =
+      PySequence_Fast(value, "must assign iterable to extended slice");
+  if (!seq) return -1;
+  if (PySequence_Size(seq) != count) {
+    PyErr_Format(PyExc_ValueError,
+                  "attempt to assign sequence of size %zd to extended slice "
+                  "of size %zd",
+                  PySequence_Size(seq), count);
+    Py_DECREF(seq);
+    return -1;
+  }
```

Comment:
I think all the exists below here are missing a decref on seq.  I think you probably want to install the `goto err` handler for this function